src/Entities/Entity.cpp

```diff
@@ -242,10 +242,17 @@ void cEntity::TakeDamage(eDamageType a_DamageType, cEntity * a_Attacker, int a_R
 {
 	TakeDamageInfo TDI;
 	TDI.DamageType = a_DamageType;
-	TDI.Attacker = a_Attacker;
+	if ((a_Attacker != nullptr) && a_Attacker->IsPawn())
+	{
+		TDI.Attacker = a_Attacker;
+	}
+	else
+	{
+		TDI.Attacker = nullptr;
+	}
```

Comment:
Alternativaly:
```
cPawn * TDI.PawnAttacker - The pawn (player or mob) which caused the damage. nullptr if no pawn is involved.
cProjectile * TDI.ProjectileAttacker - The projectile which caused the damage. nullptr if no projectile is involved.
cEntity * TDI.Attacker - DEPRECATED. If melee damage, points to the pawn. If projectile damage, points to the projectile. Otherwise a nullptr. DO NOT USE IN NEW PLUGINS.
```
And we can phase out `TDI.Attacker` in the future.